src/foam/java/refinements.js

```diff
@@ -239,7 +239,7 @@ foam.CLASS({
       // Special merging behaviour for args.
       var argCount = Math.max(this.args.length, child.args.length)
       for ( var i = 0 ; i < argCount ; i++ ) {
-        result.args[i] = this.args[i].clone().copyFrom(child.args[i]);
+        result.args[i] = (this.args.length <= i) ? child.args[i] : this.args[i].clone().copyFrom(child.args[i]);
       }
```

Comment:
Or, how about this which avoids the max() and the check on each iteration:
var i = 0;
for ( ; i < this.args.length   ; i++ ) result.args[i] = this.args[i].clone().copyFrom(child.args[i]);
for ( ; i < child.args.length ; i++ ) result.args[i] = child.args[i];